<launch>
	<!-- Publish Point Cloud From Bag -->
	<node pkg="pcl_practice" type="pcl_practice_node" name="my_node">
    <!-- Parameters -->
    <param name="debug" value="true" type="bool"/>

    <param name="applyBoxFilter" value="true" type="bool"/>
    <param name="boxFilterDist" value="5.0" type="double"/>

		<param name="applyVoxelGridFilter" value="false" type="bool"/>
    <param name="voxelGridLeafSize" value="1.0" type="double"/>

    <param name="findSurfaceNormals" value="true" type="bool"/>
    <param name="neighborRadius" value=".03" type="double"/>

    <!-- Remapping -->
    <remap from="input" to="/velodyne_points"/>
    <remap from="output" to="/example"/>
  </node>

  	<!-- Play Bag -->
	<arg name="path" default="l~/Desktop/Parallels Shared Folders/Home/Desktop/Autonomous Robotics Lab/Point Cloud Library/mine_bags/"/>
	<arg name="file" default="mine_data_2018-05-14-14-42-01_54"/>
	<node pkg="rosbag" type="play" name="player" output="screen" args="-l $(arg path)$(arg file).bag"/>

  	<!-- Visualize the point cloud in rviz -->
  	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find pcl_practice)/rviz_config/pcl_tutorial.rviz" required="true"/>
</launch>